internal/services/containerapps/container_app_environment_resource.go

```diff
@@ -304,6 +308,12 @@ func (r ContainerAppEnvironmentResource) Create() sdk.ResourceFunc {
 				managedEnvironment.Properties.VnetConfiguration.Internal = pointer.To(containerAppEnvironment.InternalLoadBalancerEnabled)
 			}
 
+			ident, err := identity.ExpandSystemAndUserAssignedMapFromModel(containerAppEnvironment.Identity)
+			if err != nil {
+				return fmt.Errorf("expanding identity: %+v", err)
+			}
+			managedEnvironment.Identity = pointer.To(identity.LegacySystemAndUserAssignedMap(*ident))
```

Comment:
We can use the helper function for legacy identity maps

```suggestion
			ident, err := identity.ExpandLegacySystemAndUserAssignedMapFromModel(containerAppEnvironment.Identity)
			if err != nil {
				return fmt.Errorf("expanding `identity`: %+v", err)
			}
			managedEnvironment.Identity = ident
```

Reply:
Will change.